app/code/community/Ebizmarts/MailChimp/Model/Observer.php

```diff
@@ -338,88 +524,67 @@ public function loadCustomerToQuote(Varien_Event_Observer $observer)
     }
 
     /**
-     * Set the products included the order to be updated on MailChimp on the next cron job run.
+     * Set the products included in the credit memo to be updated on MailChimp on the next cron job run.
      *
-     * @param  Varien_Event_Observer $observer
+     * @param Varien_Event_Observer $observer
      * @return Varien_Event_Observer
-     * @throws Exception
      */
-    public function newOrder(Varien_Event_Observer $observer)
+    public function newCreditMemo(Varien_Event_Observer $observer)
     {
-        $post = Mage::app()->getRequest()->getPost('mailchimp_subscribe');
+        $creditMemo = $observer->getEvent()->getCreditmemo();
+        $order = $creditMemo->getOrder();
+        $storeId = $order->getStoreId();
         $helper = $this->makeHelper();
-        if (isset($post)) {
-            $order = $observer->getEvent()->getOrder();
-            $email = $order->getCustomerEmail();
-            $subscriber = $helper->loadListSubscriber($post, $email);
-            if ($subscriber) {
-                $helper->subscribeMember($subscriber, true);
-            }
-        }
+        $ecomEnabled = $helper->isEcomSyncDataEnabled($storeId);
 
-        $this->removeCampaignData();
+        if ($ecomEnabled) {
 
-        $order = $observer->getEvent()->getOrder();
-        $items = $order->getAllItems();
-        foreach ($items as $item) {
-            if ($item->getProductType() == 'bundle' || $item->getProductType() == 'configurable') {
-                continue;
-            }
+            $mailchimpStoreId = $helper->getMCStoreId($storeId);
 
-            $mailchimpStoreId = $this->makeHelper()->getMCStoreId($order->getStoreId());
-            $this->makeHelper()->saveEcommerceSyncData($item->getProductId(), Ebizmarts_MailChimp_Model_Config::IS_PRODUCT, $mailchimpStoreId, null, null, 1, null, null, null, true);
-        }
+            $items = $creditMemo->getAllItems();
 
-        return $observer;
-    }
+            foreach ($items as $item) {
+                if ($item->getProductType() == 'bundle' || $item->getProductType() == 'configurable') {
```

Comment:
use `Mage_Catalog_Model_Product_Type::TYPE_BUNDLE` and `Mage_Catalog_Model_Product_Type::TYPE_CONFIGURABLE`